### PR TITLE
FX-to-ONNX exporter prototype

### PR DESCRIPTION
**Pull request by wschin**
This PR mirrors https://github.com/pytorch/pytorch/pull/88462 and enables the team to consolidate all related efforts.

As FX graph becomes the de facto IR in all new PyTorch compiler features, we want to make sure FX and ONNX can work well together. This PR is an experimental attemp to export GraphModule into ONNX. To reuse existing exporters in symbolic_opset*.py files, this prototype still heavily depends on TorchScript. I do want to remove TorchScript dependency, as long as people get time to re-write all exporters.

### CODE REVIEW

File: torch/onnx/_fx.py [hunk 1]
```diff
@@ -0,0 +1,330 @@
+# Owner(s): ["module: onnx"]
```
Comment (justinchuby) [inline torch/onnx/_fx.py:1]: I would put this file in either `_internal` or the `experimental` path. @BowenBao do you have suggestions?

Comment (justinchuby) [inline torch/onnx/_fx.py:1]: Also no need to specify owners for non-tests

File: torch/onnx/_fx.py [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+            #
+            # TODO(wechi): in the future, we might want to write individual exporter for each overload, if,
+            # for example, they have different type promotion rules. If so, just map different overloads to
+            # different exporter keys.
+            table[op_overload] = op_overload_packet._qualified_op_name
+
+    return table
+
+
+_op_overload_to_exporter_key_table = _create_op_overload_to_exporter_key_table()
```
Comment (justinchuby) [inline torch/onnx/_fx.py:57]: nit: `_UPPER_CASE` for const like globals

File: torch/onnx/verification.py [hunk 3]
```diff
@@ -727,3 +727,45 @@ def verify(
...
+    if isinstance(model, torch.nn.Module):
+        onnx_model = _fx._export_module(model, *input_args)
+    elif callable(model):
+        onnx_model = _fx._export_function(model, *input_args)
+    else:
+        raise ValueError(
+            f"model must be a torch.nn.Module or Callable but got {type(model)}"
+        )
+
+    import onnx
```
Comment (justinchuby) [inline torch/onnx/verification.py:752]: I would avoid inline import to onnx, or provide a nice error message to users on import errors.

**Comment by wschin** [comment]
@justinchuby. I am creating another PR to address your comments (will be out next hour once PyTorch builds fine locally). I will merge this PR now to enable parallel works. Many thanks!

File: torch/onnx/_fx.py (hunk 1 above)
Comment (wschin) [reply torch/onnx/_fx.py:1]: It will be in `_internal`.

File: test/onnx/test_fx_to_onnx.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                tensor_x = torch.flatten(tensor_x, 1)
+                tensor_x = self.fc1(tensor_x)
+                tensor_x = F.sigmoid(tensor_x)
+                tensor_x = self.fc2(tensor_x)
+                output = F.log_softmax(tensor_x, dim=1)
+                return output
+
+        tensor_x = torch.rand((64, 1, 28, 28), dtype=torch.float32)
+        onnx_model = _export_module(MNISTModel(), tensor_x)
+        onnx.checker.check_model(onnx_model)
```
Comment (AllenTiTaiWang) [inline test/onnx/test_fx_to_onnx.py:47]: Better use full_check:

```suggestion
        onnx.checker.check_model(onnx_model, full_check=True)
```

**Pull request merged**